feature-libs/product-configurator/rulebased/components/conflict-solver-dialog/configurator-conflict-solver-dialog.component.spec.ts

```diff
@@ -0,0 +1,127 @@
+    return of({
+      conflictGroup: of([]),
+      routerData: of(undefined),
+    });
+  }
+
+  closeDialog(_reason: string): void {}
+}
+
+describe('ConfiguratorConflictSolverDialogComponent', () => {
+  let component: ConfiguratorConflictSolverDialogComponent;
+  let fixture: ComponentFixture<ConfiguratorConflictSolverDialogComponent>;
+  let configuratorCommonsService: ConfiguratorCommonsService;
+  let launchDialogService: LaunchDialogService;
+  let mockLanguageService;
+
+  beforeEach(
+    waitForAsync(() => {
+      mockLanguageService = {
+        getAll: () => of([]),
+        getActive: jasmine.createSpy().and.returnValue(of('en')),
+      };
+
+      TestBed.configureTestingModule({
+        imports: [I18nTestingModule, IconModule],
+        declarations: [ConfiguratorConflictSolverDialogComponent],
+        providers: [
+          { provide: IconLoaderService, useClass: MockIconFontLoaderService },
+          { provide: LanguageService, useValue: mockLanguageService },
+          {
+            provide: ConfiguratorCommonsService,
+            useClass: MockConfiguratorCommonsService,
+          },
+          { provide: LaunchDialogService, useClass: MockLaunchDialogService },
+        ],
+      }).compileComponents();
+    })
+  );
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(
+      ConfiguratorConflictSolverDialogComponent
+    );
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+
+    configuratorCommonsService = TestBed.inject(
+      ConfiguratorCommonsService as Type<ConfiguratorCommonsService>
+    );
+
+    spyOn(configuratorCommonsService, 'updateConfiguration').and.callThrough();
+
+    launchDialogService = TestBed.inject(LaunchDialogService);
+    spyOn(launchDialogService, 'closeDialog').and.callThrough();
+  });
+
+  afterEach(() => {
+    component.ngOnDestroy();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  describe('updateConfiguration', () => {
+    it('should update a configuration through the facade layer ', () => {
+      const event: ConfigFormUpdateEvent = {
+        changedAttribute: undefined,
+        ownerKey: 'product/TEST_PRODUCT',
+        updateType: Configurator.UpdateType.ATTRIBUTE,
+      };
+
+      component.updateConfiguration(event);
+
+      expect(configuratorCommonsService.updateConfiguration).toHaveBeenCalled();
+      expect(
+        configuratorCommonsService.updateConfiguration
+      ).toHaveBeenCalledWith(
+        event.ownerKey,
+        event.changedAttribute,
+        event.updateType
+      );
+    });
+  });
+
+  describe('dismissModal', () => {
+    it('should close dialog when dismissModal is called', () => {
+      const reason = 'Close conflict solver dialog';
+      component.ngOnInit();
+      component.dismissModal(reason);
+      expect(launchDialogService.closeDialog).toHaveBeenCalled();
+      expect(launchDialogService.closeDialog).toHaveBeenCalledWith(reason);
```

Comment:
i think expect in line 124 is sufficient so that line 123 can be deleted

Reply:
Done!

Reply:
Done!